### PR TITLE
ldapadmin - adding logs

### PR DESCRIPTION
**Pull request by Vampouille**
Please do *not* merge this PR

### CODE REVIEW

**Comment by Vampouille** [comment]
Optimisation of ogc statistics is done in update.sql file. This file converts original `ogcstatistics.ogc_services_log` table to a partitionned version (see http://www.postgresql.org/docs/9.1/static/ddl-partitioning.html).
We have tested migration with 66M records (pigma datas), it should take around 5h. During this migration, web services should be stopped in order to prevent insertion of new records in ogc_services_log table.

**Comment by fvanderbiest** [comment]
Any update on this ? 
It's growing bigger and bigger, I fear it might become unreviewable ...

**Comment by pmauduit** [comment]
I pair-worked with @Vampouille so it's good to go for me (I'd like to review it more thoroughly once again though). In addition, it shall be feature-complete, so it won't grow any further logically.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -28,6 +28,26 @@ Version 15.12
    * drop your groups organizationalUnit (```ou```)
    * import the updated groups.ldif file.
 
+ * There is several change about structure of log in database for ogc statistic part. If your
```
Comment (pmauduit) [inline RELEASE_NOTES.md]: *changes

File: analytics/src/main/java/org/georchestra/analytics/StatisticsController.java [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+	 *
+	 * Test by group:
+	   curl -XPOST --data-binary '{"group": "ADMINISTRATOR", "startDate": "2015-01-01", "endDate": "2015-12-01" }' \
+           -H'Content-Type: application/json'   http://localhost:8280/analytics/ws/layersUsage -i
+           
+	   curl -XPOST --data-binary '{"user": "testadmin", "limit": 10, "startDate": "2015-01-01", "endDate": "2015-12-01" }' \
+           -H'Content-Type: application/json'   http://localhost:8280/analytics/ws/layersUsage -i
+           
+  	   curl -XPOST --data-binary '{"limit": 10, "startDate": "2015-01-01", "endDate": "2015-12-01" }' \
+           -H'Content-Type: application/json'   http://localhost:8280/analytics/ws/layersUsage -i
```
Comment (pmauduit) [inline analytics/src/main/java/org/georchestra/analytics/StatisticsController.java]: Note: to be removed (unecessary in the javadoc)

File: analytics/src/main/java/org/georchestra/analytics/StatisticsController.java [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+		return new JSONObject().put("results", results)
+				.toString(4);
+	}
+	
+	/**
+	 * 
+	 * @param payload the JSON object containing the parameters
+	 * @param response the HTTP Servlet Response object, used to set the 40x HTTP code in case of errors.
+	 *
+	 * curl -XPOST --data-binary '{"group": "ADMINISTRATOR", "startDate": "2015-01-01", "endDate": "2015-12-01" }' \
```
Comment (pmauduit) [inline analytics/src/main/java/org/georchestra/analytics/StatisticsController.java]: same

File: analytics/src/main/java/org/georchestra/analytics/StatisticsController.java [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+	}
+	
+
+	/** Granularity used for the returned date type in combined requests statistics */
+	public static enum GRANULARITY { HOUR, DAY, WEEK, MONTH }
+
+	/**
+	 * Testing onto users:
+	 *
+	 *  curl -XPOST --data-binary '{"user": "testadmin", "startDate": "2015-01-01", "endDate": "2015-12-01" }' \
```
Comment (pmauduit) [inline analytics/src/main/java/org/georchestra/analytics/StatisticsController.java]: `curl ...` to be removed

File: ldapadmin/src/main/webapp/WEB-INF/spring/webmvc-config.xml [hunk 5]
```diff
@@ -81,18 +88,22 @@
...
   <bean class="org.georchestra.ldapadmin.ws.backoffice.users.UserRule">
     <property name="listOfprotectedUsers">
-      <description>Add one or more user identifiers (uid) of protected user</description>
-      <list>
-        <value>geoserver_privileged_user</value>
-        <!--  to define an other uid you should add a value like is shown in the following example-->
-        <!--  <value>${protectedUser.uid2}</value> -->
-      </list>
+      <description>Comma separated list of one or more user identifiers (uid) of protected user</description>
+        <!-- Users are defined as a comma separated list of uid and can be override in data dir with "protectedUserList" key-->
```
Comment (pmauduit) [inline ldapadmin/src/main/webapp/WEB-INF/spring/webmvc-config.xml]: overridden

**Comment by fvanderbiest** [comment]
What's left to do for this ? Is it ready to merge ?

**Comment by fvanderbiest** [comment]
"analytics - improve performance on big logs #556" depends on this one.
It's high time we merge this for 15.12. Can we ?

**Comment by Vampouille** [comment]
Yes this can be merged.

**Pull request merged**